### DISCUSSION TITLE
How to use realtime...

### DISCUSSION THREAD

**Discussion by johnbrett** [supabase/supabase#2101] [Questions] [answered]
I'm looking to try out the realtime functionality, covered in the docs here: https://supabase.io/docs/reference/javascript/subscribe#notes

I want to get the next.js example working (https://github.com/supabase/realtime/tree/master/examples/next-js), but with a hosted supabase database and realtime server, instead of local ones.

How do you connect to this, it doesn't seem to use the supabase url - is there a hosted realtime server like there is API server, and if so what is it? 

I see in the server docs 

> Authorizing Client Connection: You can pass in the JWT by following the instructions under Usage in the @supabase/realtime-js 
client library or as query param in the WebSocket URL (e.g. wss://abc.supabase.co/realtime/v1/websocket?vsn=1.0.0&apikey=jwt).

is `abc` here the regular supabase url? I've tried this and variations for the realtime client url connection string, e.g. 

```
new RealtimeClient(wss://abc.supabase.co)
new RealtimeClient(wss://abc.supabase.co/realtime/v1)
```
But these fail to connect. If you do need to run the server locally, would be worth mentioning in the docs, this is a big DX hurdle as my app is hosted on Vercel, I'd rather not spin up a server somewhere to run the realtime server.

**Comment by easy-street** [comment, marked as answer]
it is not required to host a server locally. hosted supabase projects include a realtime endpoint that works well with apps deployed to Vercel.
 
the example in the `supabase/realtime` repo demonstrates a low-level approach. instead,  take a look at the [nextjs-slack-clone example](https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/lib/Store.js#L33), which implements a realtime chat app with [`@supabase/supabase-js`](https://github.com/supabase/supabase-js) using [simple APIs](https://supabase.io/docs/reference/javascript/subscribe) for working with realtime functionality in supabase projects.

it is possible to run the the nextjs example in the `supabase/realtime` repo with hosted supabase projects by initializing the `RealtimeClient` with the anon API key. for example:
```
new RealtimeClient("wss://abc.supabase.co/realtime/v1", {
  params: { apikey: "<anon API key>" },
})
```

this is not necessary in supabase projects implemented with `@supabase/supabase-js`, which handles [initializing the realtime client](https://github.com/supabase/supabase-js/blob/master/src/SupabaseClient.ts#L158).

**Reply by johnbrett** [reply]
Wow - this answers everything, thank you!! Not sure how I ended up down the rabbit hole of the realtime project, might be worth adding notes in readme there that this is a low level API and you likely don't want to be there unless looking to work directly on the realtime server.
